src/programs/state/massachusetts/snap.js

```diff
@@ -1,14 +1,443 @@
+//   'currentOtherMedicalCostsMonthly',
+//   'currentRentShareMonthly',
+//   'currentContractRentMonthly',
+//   'currentRentMonthly'
+//   */
+// ];
+
+const getAllowance = function(client, timeframe, data, baseRate ) {
+  if ( client[timeframe + 'HouseholdSize'] > 8  ) {
+    return data[8] + (client[timeframe + 'HouseholdSize'] - 8) * baseRate;
+  } else {
+    return data[client[timeframe + 'HouseholdSize']];
+  }
+};
+
+//GROSS INCOME TEST
+const hasDisabledOrElderlyMember = function (client, timeframe) {
+  return client[timeframe + 'DisabledOrElderlyMember'];
+};
+
+const getTotalMonthlyGross = function (client, timeframe) {
+  // return getSimpleGrossIncomeMonthly(client, timeframe) - toCashflow(client, timeframe, 'ChildSupportPaidOut');
+  return toCashflow(client, 'future', 'EarnedIncome') + getGrossUnearnedIncomeMonthly(client, timeframe) - toCashflow(client, timeframe, 'ChildSupportPaidOut');
+};
+
+const getPovertyGrossIncomeLevel = function (client, timeframe ) {
+  return getAllowance(client, timeframe, data.povertyGrossIncome, data.overNumberHouseholdRate);
+};
+
+const checkIncome = function (client, timeframe) {
+  var totalMonthlyGross = getTotalMonthlyGross(client, timeframe);
+  var povertyGrossIncomeLevel = getPovertyGrossIncomeLevel(client, timeframe);
+  var isDisabledOrElderlyMember = hasDisabledOrElderlyMember(client, timeframe);
+  if ((totalMonthlyGross > povertyGrossIncomeLevel) && isDisabledOrElderlyMember) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+// const isAssetTest = function(client, timeframe) {
+//   if (checkIncome(client, timeframe)) {
+//     return true; // Yes, "Yes, assets must be <=$3,250"
+//   } else {
+//     return false; //No
+//   }
+// };
+
+const isNetIncomeTest = function(client, timeframe) {
+  if (checkIncome(client, timeframe)) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+const getGrossIncomeTestResult = function (client, timeframe) {
+  var totalMonthlyGross = getTotalMonthlyGross(client, timeframe);
+  var povertyGrossIncomeLevel = getPovertyGrossIncomeLevel(client, timeframe);
+  var isPassGrossIncomeTest = null;
+  if ( hasDisabledOrElderlyMember(client, timeframe) ) {
+    isPassGrossIncomeTest = true;
+  } else {
+    // TODO: must double checked in the documentation. Two different results in both excel calculator and website calculator
+    // minor difference "<" in website calculator logic on line 469.
+    if ( totalMonthlyGross <= povertyGrossIncomeLevel ) {
+      isPassGrossIncomeTest = true;
     } else {
-        return {result: 'warning', details: `Your income puts you at ${percentPov.toFixed()}% of the federal poverty level, which is above the 80% limit.`, benefitValue: 0};
+      isPassGrossIncomeTest = false;
     }
-}
+  }
+  return isPassGrossIncomeTest;
+};
+
+// INCOME DEDUCTIONS
+const getStandardDeduction = function (client, timeframe) {
+  if (client[timeframe + 'HouseholdSize'] >= 6) {
+    return data.standardDeduction[6];
```

Comment:
I think you can use `getMaxIntKey()` from `helpers/get-gov-data.js` here (you'll have to change it to export `getMaxIntKey` though.)